src/script/page/AppMain.tsx

```diff
@@ -17,38 +17,45 @@
  *
  */
 
-import {FC, useEffect, useLayoutEffect} from 'react';
+import {FC, useCallback, useEffect, useMemo} from 'react';
 
+import {ConnectionStatus} from '@wireapp/api-client/lib/connection/';
 import {amplify} from 'amplify';
 import {container} from 'tsyringe';
 
+import {Runtime} from '@wireapp/commons';
 import {StyledApp, THEME_ID, useMatchMedia} from '@wireapp/react-ui-kit';
 import {WebAppEvents} from '@wireapp/webapp-events';
 
 import {CallingContainer} from 'Components/calling/CallingOverlayContainer';
 import {GroupCreationModal} from 'Components/Modals/GroupCreation/GroupCreationModal';
 import {LegalHoldModal} from 'Components/Modals/LegalHoldModal/LegalHoldModal';
+import {useLegalHoldModalState} from 'Components/Modals/LegalHoldModal/LegalHoldModal.state';
 import {PrimaryModal} from 'Components/Modals/PrimaryModal';
 import {PrimaryModalComponent} from 'Components/Modals/PrimaryModal/PrimaryModal';
 import {showUserModal, UserModal} from 'Components/Modals/UserModal';
+import {iterateItem} from 'Util/ArrayUtil';
 import {useKoSubscribableChildren} from 'Util/ComponentUtil';
+import {t} from 'Util/LocalizerUtil';
+import {matchQualifiedIds} from 'Util/QualifiedId';
+import {isConversationEntity} from 'Util/TypePredicateUtil';
 
 import {AppLock} from './AppLock';
 import {LeftSidebar} from './LeftSidebar';
 import {MainContent} from './MainContent';
 import {PanelEntity, PanelState, RightSidebar} from './RightSidebar';
-import {RootProvider} from './RootProvider';
 import {useAppMainState, ViewType} from './state';
-import {useAppState, ContentState} from './useAppState';
+import {useAppState, ContentState, ListState} from './useAppState';
```

Comment:
This file is becoming way to big. Please try to reorganize this into multiple components that have limited responsibilities. I'm sure we can find something more suitable :)